### PR TITLE
AKCORE-208: Move TestUtils to scala and call producer.close() in ShareConsumerTest

### PR DESCRIPTION
**Pull request by ShivsundarR**
*What*
Move TestUtils to scala and call producer.close() in ShareConsumerTest

### CODE REVIEW

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 1]
```diff
@@ -330,6 +334,7 @@ public void testHeaders() {
             if (header != null)
                 assertEquals("headerValue", new String(header.value()));
         }
+        producer.close();
```
Comment (AndrewJSchofield) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java:337]: Need to close the consumer here too.

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java [hunk 2]
```diff
@@ -344,6 +349,7 @@ private void testHeadersSerializeDeserialize(Serializer<byte[]> serializer, Dese
 
         List<ConsumerRecord<byte[], byte[]>> records = consumeRecords(shareConsumer, numRecords);
         assertEquals(numRecords, records.size());
+        producer.close();
```
Comment (AndrewJSchofield) [inline core/src/test/java/kafka/test/api/ShareConsumerTest.java:352]: And again. Please check all instances of share consumer are closed.

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]
I think you caught almost all of them. Good job. Also need to close the producer in `testAcknowledgeCommitCallbackCallsShareConsumerDisallowed`.

File: core/src/test/java/kafka/test/api/ShareConsumerTest.java (hunk 2 above)
Comment (ShivsundarR) [reply core/src/test/java/kafka/test/api/ShareConsumerTest.java:354]: Okay, I have updated the code and pushed again now.

**Pull request merged**